### PR TITLE
CWL version of workflow

### PR DESCRIPTION
**Pull request by wshands**

### CODE REVIEW

File: samtools/metrics.cwl [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    s:identifier: https://orcid.org/0000-0001-5173-4627
+    s:email: jshands@ucsc.edu
+    s:name: Walter Shands
+
+s:codeRepository: https://github.com/wshands/dockstore-training
+s:dateCreated: "2020-06-22"
+s:license: https://spdx.org/licenses/Apache-2.0
+
+s:keywords: edam:topic_0091 , edam:topic_0622
+s:programmingLanguage: Python
```
Comment (Ldcabansay) [inline samtools/metrics.cwl:19]: samtools is written in C, but i don't think we necessarily need to specify it

File: test.cwl.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "class": "File",
+      "path": "/Users/louisecabansay/gitOrgs/dockstore-training/data/NA12878_chr20_ds_r2.fq"
+  },
+  "ref_fasta": {
+       "class": "File",
+       "path": "/Users/louisecabansay/gitOrgs/dockstore-training/data/ref/ref_hg37_chr20.fa"
+  },
+  "ref_fasta_fai": {
+       "class": "File",
+       "path", "/Users/louisecabansay/gitOrgs/dockstore-training/data/ref/ref_hg37_chr20.fa.fai"
```
Comment (Ldcabansay) [inline test.cwl.json:19]: colon

File: samtools/metrics.cwl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    doc: SAM file to analyze.
+
+outputs:
+  flagstat_metrics:
+    type: File
+    outputBinding:
+      glob: "*.metrics"
+    doc: Metrics on the input SAM file.
+
+baseCommand: [flagstat.sh]
```
Comment (Ldcabansay) [inline samtools/metrics.cwl:61]: For the workshop,  to keep things simple I think we should be putting the command directly in the CWL.
Though I think we can keep the script file itself. We can have another version that uses calling to the script to show wrap commands for more complex cases.

File: samtools/Dockerfile [hunk 4]
```diff
@@ -27,6 +27,8 @@ RUN wget https://github.com/samtools/samtools/releases/download/1.10/samtools-1.
 	./configure --prefix $(pwd) && \
 	make
 
+COPY ./flagstat.sh /usr/local/bin
```
Comment (Ldcabansay) [inline samtools/Dockerfile:30]: I dont think this should be part of the docker file as the script calls the flagstat command itself. 
The dockerfile should just set up the environment with samtools

File: bwa/Dockerfile [hunk 5]
```diff
@@ -14,3 +14,6 @@ USER root
 # install packages you want available in container
 RUN apt update
 RUN apt install bwa
+
+COPY bwa_align.sh /usr/local/bin/
```
Comment (Ldcabansay) [inline bwa/Dockerfile:18]: not seeing this script checked-in, but same comment as the samtools dockerfile with flatstat.sh

File: samtools/metrics.cwl (hunk 1 above)
Comment (wshands) [reply samtools/metrics.cwl]: Removed excess metadata

File: test.cwl.json (hunk 2 above)
Comment (wshands) [reply test.cwl.json]: fixed

File: bwa/Dockerfile (hunk 5 above)
Comment (wshands) [reply bwa/Dockerfile]: Removed scripts

File: samtools/Dockerfile (hunk 4 above)
Comment (wshands) [reply samtools/Dockerfile]: Removed scripts

**Comment by wshands** [comment]
Had to put the threads in the BWA options, otherwise it complicated the CWL

File: samtools/metrics.cwl (hunk 3 above)
Comment (wshands) [reply samtools/metrics.cwl]: I removed the scripts

**Comment by Ldcabansay** [comment]
Thanks looks great! Can you add more comments, like to each section? Since this is going to be used to teach people who are new to CWL and might have limited coding experience in general, it'd be helpful to explain each CWL component. For examples, I added comments to the WDL one and Natalie did it for nextflow as well.

In most ways CWL is pretty intuitive with naming conventions, so more on whether params need to be typed etc. For less straightforward pieces, short comments on what are nampaces, dct:creator, etc and why to use them if they're optional.

**Comment by wshands** [comment]
Hi Louise, I cannot merge the PR so you will have to do it.

**Pull request merged**